tests/smoke_tests/run_smoke_test.py

```diff
@@ -0,0 +1,180 @@
+logger = logging.getLogger()
+
+
+async def run_smoke_test(
+    config_path: str = "tests/smoke_tests/config.yaml",
+    dataset_path: str = "examples/datasets/mnist_data/",
+) -> None:
+    with open(config_path, "r") as file:
+        config = yaml.safe_load(file)
+
+    _preload_dataset(dataset_path, config)
+
+    # Start the server, divert the outputs to a server file
+    logger.info("Starting server...")
+
+    server_process = await asyncio.create_subprocess_exec(
+        "nohup",
+        "python",
+        "-m",
+        "examples.fedprox_example.server",
+        "--config_path",
+        config_path,
+        stdout=asyncio.subprocess.PIPE,
+        stderr=asyncio.subprocess.STDOUT,
+    )
+
+    # reads lines from the server output in search of the startup log message
+    # times out after 20s of inactivity if it doesn't find the log message
+    full_server_output = ""
+    startup_message = "FL starting"
+    output_found = False
+    while True:
+        try:
+            assert server_process.stdout is not None, "Server's process stdout is None"
+            server_output_in_bytes = await asyncio.wait_for(server_process.stdout.readline(), 20)
+            server_output = server_output_in_bytes.decode()
+            full_server_output += server_output
+        except asyncio.TimeoutError:
+            logger.error(f"Timeout waiting for server startup message '{startup_message}'")
+            break
+        logger.debug(f"Server output: {server_output}")
+        if startup_message in server_output:
+            output_found = True
+            break
+
+    assert output_found, f"Startup log message '{startup_message}' not found in server output."
+
+    logger.info("Server started")
+
+    # Start n number of clients and capture their process objects
+    client_processes = []
+    for i in range(config["n_clients"]):
+        logger.info(f"Starting client {i}")
+        client_process = await asyncio.create_subprocess_exec(
+            "nohup",
+            "python",
+            "-m",
+            "examples.fedprox_example.client",
+            "--dataset_path",
+            dataset_path,
+            stdout=asyncio.subprocess.PIPE,
+            stderr=asyncio.subprocess.STDOUT,
+        )
+        client_processes.append(client_process)
+
+    # Collecting the clients output when their processes finish
+    full_client_outputs = []
+    for i in range(len(client_processes)):
+        full_client_outputs.append(
+            await _wait_for_process_to_finish_and_retrieve_logs(client_processes[i], f"Client {i}")
+        )
+
+    logger.info("All clients finished execution")
+
+    # Collecting the server output when its process finish
+    full_server_output = await _wait_for_process_to_finish_and_retrieve_logs(server_process, "Server")
+
+    logger.info("Server has finished execution")
+
+    # server assertions
```

Comment:
Please check the assertions below and let me know if they look correct, if there is anything I should remove or anything else I should add.

Reply:
I think these look good to me. For this first pass I think these are awesome. 

For a follow up ticket, I think one thing that would be great, is if we set random seeds in the various places (potentially using Sana's module she has a PR up for right now) to make training deterministic. Then we can also check that the output metrics are as expected. This would catch issues that don't break the functionality, but might have altered (in an unexpected way) the way training happens.

Reply:
That's a great idea. I'll keep an eye on that PR and create a ticket.